comp/metadata/inventorychecks/inventorychecksimpl/inventorychecks_test.go

```diff
@@ -97,116 +97,140 @@ func TestGetInstanceMetadata(t *testing.T) {
-	assert.Equal(t, "test: 22", check1Instance2["instance_config"])
-
-	assert.Len(t, p.Metadata["check2"], 1) // check2 has one instance
-	check2Instance1 := p.Metadata["check2"][0]
-	assert.Equal(t, "check2_instance1", check2Instance1["config.hash"])
-	assert.Equal(t, "provider2", check2Instance1["config.provider"])
-	assert.Equal(t, "{}", check2Instance1["init_config"])
-	assert.Equal(t, "{}", check2Instance1["instance_config"])
-
-	// Check that metadata linked to non-existing check were deleted
-	assert.NotContains(t, "non_running_checkid", ic.data)
-
-	// Check the log sources part of the metadata
-	assert.Len(t, p.LogsMetadata, 1)
-	actualSource, found := p.LogsMetadata["redisdb"]
-	assert.True(t, found)
-	assert.Len(t, actualSource, 1)
-	expectedSourceConfig := `{"type":"file","path":"/var/log/redis/redis.log","service":"awesome_cache","source":"redis","tags":["env:prod"]}`
-	assert.Equal(t, expectedSourceConfig, actualSource[0]["config"])
-	expectedSourceStatus := map[string]string{
-		"status": "error",
-		"error":  "Error: No such file or directory",
+	for _, invChecksCfgEnabled := range []bool{true, false} {
+		testName := fmt.Sprintf("inventories_checks_configuration_enabled=%t", invChecksCfgEnabled)
+		t.Run(testName, func(t *testing.T) {
+			cInfo := []check.Info{
+				check.MockInfo{
+					Name:         "check1",
+					CheckID:      checkid.ID("check1_instance1"),
+					Source:       "provider1",
+					InitConf:     "",
+					InstanceConf: "{\"test\":21}",
+				},
+				check.MockInfo{
+					Name:         "check1",
+					CheckID:      checkid.ID("check1_instance2"),
+					Source:       "provider1",
+					InitConf:     "",
+					InstanceConf: "{\"test\":22}",
+				},
+				check.MockInfo{
+					Name:         "check2",
+					CheckID:      checkid.ID("check2_instance1"),
+					Source:       "provider2",
+					InitConf:     "{}",
+					InstanceConf: "{}",
+				},
+			}
+
+			overrides := map[string]any{
+				"inventories_configuration_enabled":        true,
+				"inventories_checks_configuration_enabled": invChecksCfgEnabled,
+			}
+
+			mockColl := collector.NewMock(cInfo)
+			mockColl.On("AddEventReceiver", mock.AnythingOfType("EventReceiver")).Return()
+			mockColl.On("MapOverChecks", mock.AnythingOfType("func([]check.Info)")).Return()
+
+			// Setup log sources
+			logSources := sources.NewLogSources()
+			src := sources.NewLogSource("redisdb", &logConfig.LogsConfig{
+				Type:       logConfig.FileType,
+				Path:       "/var/log/redis/redis.log",
+				Identifier: "redisdb",
+				Service:    "awesome_cache",
+				Source:     "redis",
+				Tags:       []string{"env:prod"},
+			})
+			// Register an error
+			src.Status.Error(fmt.Errorf("No such file or directory"))
+			logSources.AddSource(src)
+			mockLogAgent := fxutil.Test[optional.Option[logagent.Mock]](
+				t, logagent.MockModule(), core.MockBundle(), inventoryagent.MockModule(),
+			)
+			logsAgent, _ := mockLogAgent.Get()
+			logsAgent.SetSources(logSources)
```

Comment:
Nit: Looks like everything in here except `overrides` is the same between tests. Maybe we can move those except `overrides`  above the `range []bool{true, false}`

What do you think?